gnovm/pkg/gnolang/types.go

```diff
@@ -2098,6 +2108,44 @@ func assertSameTypes(lt, rt Type) {
 	}
 }
 
+func isEqualityTypes(lt, rt Type) bool {
```

Comment:
Equality is already defined as Type.TypeID() == Other.TypeID(). This guarantees an exact match, which is what is required of == equality of interface variables (but not required otherwise). 

The below snippet demonstrates in Go the behavior we want; to return false, even though the underlying base types might be compatible. 

```go
package main

import (
        "fmt"
        "reflect"
)

// implicit type conversion and equality
func main() {
        { // case 1 with slices
                type X []byte
                var x interface{} = []byte("foo")
                var xx interface{} = X(x.([]byte))
                fmt.Println(reflect.TypeOf(x), x)  // []uint8 [102 111 111]
                fmt.Println(reflect.TypeOf(xx), x) // main.X [102 111 111]
                fmt.Println(x == xx)               // false
        }

        { // case 2 with arrays
                type X [3]byte
                var x interface{} = [3]byte{102, 111, 111}
                var xx interface{} = X(x.([3]byte))
                fmt.Println(reflect.TypeOf(x), x)  // [3]uint8 [102 111 111]
                fmt.Println(reflect.TypeOf(xx), x) // main.X [102 111 111]
                fmt.Println(x == xx)               // false still.
        }
}
```

Reply:
this works fine on my branch.

Reply:
this logic has been moved to another PR, #1890.